gossipd/gossipd.c

```diff
@@ -402,6 +402,49 @@ static u8 *handle_node_announce(struct peer *peer, const u8 *msg)
 	return err;
 }
 
+/*~ Large peers often turn off gossip msgs (using timestamp_filter) from most
+ * of their peers, however if the gossip is about us, we should spray it to
+ * everyone whether they've set the filter or not, otherwise it might not
+ * propagate! */
+void push_gossip(struct daemon *daemon, const u8 *msg TAKES)
+{
+	struct peer *peer;
+
+	list_for_each(&daemon->peers, peer, list)
+		queue_peer_msg(peer, msg);
```

Comment:
Sorry for my "too early" ACK.
If we call `push_gossip(daemon, take(msg))`, here `msg` is regarded as `TAKEN` not only in `push_gossip()`, but also in `queue_peer_msg()`.
In this loop, if the parameter `msg` is marked as `TAKEN`, then the first `queue_peer_msg` would `free` it.
So, maybe `msg` shouldn't be declared as "TAKES" in `void push_gossip(struct daemon *daemon, const u8 *msg TAKES)`.
Is this reason accurate?

Reply:
Yep, good catch. Will fix...